### PR TITLE
Add support for index entrypoints, update build steps

### PR DESCRIPTION
**Pull request by joshblack**
Uses (and enhances) `scss-generator` in build scripts for all Sass projects. Also adds `index.scss` entrypoints to all packages.

#### Changelog

**New**

- `package-name.scss` and `index.scss` entrypoints for all packages
- Add support for more language grammar in scss-generator

**Changed**

- Build processes have been refactored to use `scss-generator`

**Removed**

### CODE REVIEW

File: packages/colors/docs/sass.md [hunk 1]
```diff
@@ -281,110 +280,6 @@ Define color variables
 
 ```scss
 @mixin carbon--colors() {
-  $carbon--black-100: #000000 !default !global;
```
Comment (joshblack) [inline packages/colors/docs/sass.md]: Seems like something may be missing here in the build step

File: packages/themes/scss/_tokens.scss [hunk 2]
```diff
@@ -5,322 +5,269 @@
...
-
 @import './theme-maps';
-
 /// Primary interactive color; Primary buttons
 /// @type Color
 /// @access public
 /// @group @carbon/themes
-$interactive-01: map-get($carbon--theme, interactive-01) !default;
-
+$interactive-01: map-get($carbon--theme, 'interactive-01');
```
Comment (asudoh) [inline packages/themes/scss/_tokens.scss]: Removal of `!default` here seems not intentional, is it correct...?

Comment (joshblack) [reply packages/themes/scss/_tokens.scss]: Thanks @asudoh! Will make sure to add in 😄

File: packages/colors/docs/sass.md [hunk 3]
```diff
@@ -604,7 +499,7 @@ Define color variables
       90: #003137,
       100: #081a1c,
     ),
-    'warmGray': (
+    warmGray: (
```
Comment (joshblack) [inline packages/colors/docs/sass.md]: Should this be `warm-gray` instead?

File: packages/themes/scss/_mixins.scss [hunk 4]
```diff
@@ -5,12 +5,10 @@
 // This source code is licensed under the Apache-2.0 license found in the
 // LICENSE file in the root directory of this source tree.
 //
-
 @import './theme-maps';
-
 /// Define theme variables from a map of tokens
 /// @access public
-/// @param {Map} $theme [$carbon--theme] - Map of theme tokens
+/// @param {Map} $theme [carbon--theme] - Map of theme tokens
```
Comment (joshblack) [inline packages/themes/scss/_mixins.scss]: Bug in generator

File: packages/colors/docs/sass.md [hunk 5]
```diff
@@ -146,7 +145,7 @@ Define color variables
       90: #061f80,
       100: #051243,
     ),
-    'coolGray': (
```
Comment (mattrosno) [inline packages/colors/docs/sass.md]: Were these gray re-namings intentional? E.g. `coolGray` to `cool-gray`?

Comment (joshblack) [reply packages/colors/docs/sass.md]: Yeah, we should do `param-case` in Sass, I think.

Comment (mattrosno) [reply packages/colors/docs/sass.md]: Is that a breaking change?

Comment (joshblack) [reply packages/colors/docs/sass.md]: Yeah, I guess it would be. I'll add in support for both forms for now.

**Comment by joshblack** [comment]
@mattrosno done!

**Comment by mattrosno** [comment]
@joshblack thanks! I'll approve after CI passes.

**Comment by joshblack** [comment]
@mattrosno done!

**Pull request merged**